src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -182,6 +182,17 @@ public Dataset retrieve(UUID id) {
     return datasetDao.retrieve(id);
   }
 
+  /**
+   * Fetch existing Dataset object populated only as necessary to facilitate data ingestion (i.e.
+   * without relationships or assets which are constructed via costly database calls).
+   *
+   * @param id in UUID format
+   * @return a Dataset object
+   */
+  public Dataset retrieveForIngest(UUID id) {
```

Comment:
Now that we have a way to get a dataset without relationships, could we put this dataset in the flight map? It looked like the problem with serialization before was due to the relationship structure, due its recursive definition.

Reply:
I think we can.  There were concerns raised in Datahood Design Discussions that loading the dataset into the flight map could be problematic if the dataset changes between its storage in the map and its usage later on in the flight… but we could guard against that by loading it into the working map after locking the dataset.

I thought this would be worth pursuing if / when we modify the dataset types as discussed below.